### PR TITLE
Add default searches to new users

### PR DESCRIPTION
**Pull request by BenJam**
Adds three default 'pinned searches' to new user accounts:

- Archivable: any closed or merged issues or PRs that are not yet in the archive
- Mergeable: any open pull request that is currently passing CI 
- My PRs: any PR that is in your inbox for which you were the author

The aim here is to get every new user _aware_ of pinned searches and to help onboard them quickly by removing all the chaff (archivable) focussing on quick wins (mergeable) and then getting focussed on their own work (my prs).

I'm not sure how best to roll this across the 12k users we have on Octobox.io. Some help there would be 👌

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -161,4 +163,11 @@ def can_comment?(subject)
     return true if github_app_authorized? && subject.repository.commentable?
     return false
   end
+
+  def create_default_pinned_searches 
+    pinned_searches.new(query: 'state:closed,merged archived:false', name: 'Archivable')
+    pinned_searches.new(query: 'type:pull_request state:open status:success', name: 'Mergeable')
+    pinned_searches.new(query: "type:pull_request author:#{github_login} inbox:true", name: 'My PRs')
+    save
```
Comment (andrew) [inline app/models/user.rb]: This just calls save on the user, not the previously initialized objects

**Comment by andrew** [comment]
For existing users, perhaps we can create a rake task that checks if they have any pinned searches, and if not calls `create_default_pinned_searches` on them, otherwise we can just run it in the production console.

File: app/models/user.rb [hunk 2]
```diff
@@ -160,4 +162,10 @@ def can_comment?(subject)
     return true if github_app_authorized? && subject.repository.commentable?
     return false
   end
+
+  def create_default_pinned_searches 
+    pinned_searches.create(query: 'state:closed,merged archived:false', name: 'Archivable')
+    pinned_searches.create(query: 'type:pull_request state:open status:success', name: 'Mergeable')
```
Comment (pradyunsg) [inline app/models/user.rb]: Should this also include archived:false?

Comment (andrew) [reply app/models/user.rb]: Fixed, good spot!

Comment (BenJam) [reply app/models/user.rb]: ++ thanks @pradyunsg

**Pull request merged**